File: src/main/java/by/iba/repair_report/service/UserService.java
package by.iba.repair_report.service;

import by.iba.repair_report.dto.request.SignupRequest;
import by.iba.repair_report.dto.response.UserDto;
import by.iba.repair_report.entity.User;

import java.util.List;

public interface UserService {
    User createUser(SignupRequest signUpRequest);
    UserDto updateUser(Long userId, SignupRequest signUpRequest);
    void deleteUser(Long userId);
    List<UserDto> getAllUsers();
    UserDto getUserById(Long userId);
    void changePassword(Long userId, String newPassword);
    UserDto getCurrentUser();
    boolean hasAccessToBranch(Long userId, Long branchId);
    boolean hasAccessToEnterprise(Long userId, Long enterpriseId);
}

File: src/main/frontend/src/services/userService.js
import api from './api.js';

export const userService = {
    getAllUsers: async () => {
        const response = await api.get('/admin/users');
        return response.data;
    },

    getUserById: async (id) => {
        const response = await api.get(`/admin/users/${id}`);
        return response.data;
    },

    createUser: async (userData) => {
        const response = await api.post('/admin/users', userData);
        return response.data;
    },

    updateUser: async (id, userData) => {
        const response = await api.put(`/admin/users/${id}`, userData);
        return response.data;
    },

    deleteUser: async (id) => {
        await api.delete(`/admin/users/${id}`);
    },

    changePassword: async (id, passwordData) => {
        await api.put(`/admin/users/${id}/password`, passwordData);
    }
};